mousebender/simple.py

```diff
@@ -141,6 +143,19 @@ class ArchiveLink:
 class _ArchiveLinkHTMLParser(html.parser.HTMLParser):
     def __init__(self):
         self.archive_links = []
+
+        self.index: Dict[
```

Comment:
I've laid out the intended struct here. What would a user expect to see in terms of documentation? Should I add a description of the struct returned from `parse_archive_links` and some sample use cases? I am inclined to do so, but perhaps that is too pedantic 🤔.

Reply:
To be clear, there's no _struct_ (a [struct](https://docs.python.org/3/library/struct.html#module-struct) has different meaning in Python). You're returning a dict with a bunch of stuff. 😄 

Documentation will be required to explain the structure., else folks won't know how to traverse it.

Reply:
Is there a reason to do this as part of the parse instead of as a separate step? If there's a perf bump for doing it here then great, but otherwise composability and flexibility would suggest getting the archive links in a page is a separate step from organizing them.